lib/vsc/mympirun/common.py

```diff
@@ -96,8 +96,18 @@ def sched_to_key(klass):
 
     # next, try to use the scheduler defined by environment variables
     for sched in found_sched:
-        nodeinfo = not hasattr(sched, 'SCHED_ENVIRON_NODE_INFO') or sched.SCHED_ENVIRON_NODE_INFO in os.environ
-        if nodeinfo and sched.SCHED_ENVIRON_ID in os.environ:
+
+        # take into account that SCHED_ENVIRON_NODE_INFO can be None,
+        # and chechking "None in os.environ" fails hard in Python 3 (string value is required)
+        if hasattr(sched, 'SCHED_ENVIRON_NODE_INFO'):
```

Comment:
should be 
```
nodeinfo = getattr(sched, 'SCHED_ENVIRON_NODE_INFO', True) and (sched.SCHED_ENVIRON_NODE_INFO or '') in os.environ
```

Reply:
This is not correct, tests fail now, and it makes sense: if `SCHED_ENVIRON_NODE_INFO` is not an attribute, then you get `True and ...` where the `...` does a `sched.SCHED_ENVIRON_NODE_INFO` which fails hard.